lib/maintenance/MaintenanceManager.js

```diff
@@ -67,6 +67,17 @@ class MaintenanceManager {
       .findOne(CONST.DB_MODEL.MAINTENANCE_DETAIL, criteria);
   }
 
+  getDowntimePhase(maintenanceInfo, downTimePhases) {
```

Comment:
if I understand correctly , `maintenanceInfo.progress` is an array which is filled up the way we have been updating maintenance , correct ?

Ideally we should check only for the last entry of that array and check if matches with any of the the `downTimePhases` right ?

Reply:
yes you are right in a way, but the issue is its truly not a 'phase' info that we have. From progress info  we are inferring the phases. When updating the phase both start and end of the phases are logged in. So if we just check for the last entry then there could be some (corner) cases where we might be in a stage where in broker update is succeeded and we might just see that as not in maintenance and start the job. So just to avoid this having this. But yeah, if you insist i can just check for last entry or last but 1 entry which should take care even of the boundary case.

Reply:
yes ok..